src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Lower.java

```diff
@@ -3518,16 +3518,16 @@ private void visitIterableForeachLoop(JCEnhancedForLoop tree) {
                                               syms.iterableType.tsym);
             if (iterableType.getTypeArguments().nonEmpty())
                 iteratorTarget = types.erasure(iterableType.getTypeArguments().head);
-            Type eType = types.skipTypeVars(tree.expr.type, false);
-            tree.expr.type = types.erasure(eType);
-            if (eType.isCompound())
+            tree.expr.type = types.erasure(types.skipTypeVars(tree.expr.type, false));
```

Comment:
Shouldn't types already be erased by now (Lower runs after TransTypes?)

Reply:
Perhaps (actually very likely). This transformation is just simplifying existing code.